UPGRADING_DEPENDENCIES.md

```diff
@@ -0,0 +1,103 @@
+# Guide to Upgrading Dependencies
+
+Upgrading Go or Node.js requires making changes in many different files. See below for a list and explanation for each.
+
+## Go
+
+- CircleCi
+- `grafana/build-container`
+- Appveyor
+- Dockerfile
+
+## Node.js
+
+- `package.json`
```

Comment:
I don't think we specify the node-version in package.json today, if it's the "engines" setting you mean (https://docs.npmjs.com/files/package.json#engines).

Maybe we should?